twisted/internet/interfaces.py

```diff
@@ -2256,6 +2256,28 @@ def acceptableProtocols():
 
 
 
+class ITLSContextFactory(Interface):
```

Comment:
This is a good idea. However, ITLSOpenSSLContextFactory is overlong and redundant, so I'm going to change the name to IOpenSSLContextFactory.